### PR TITLE
feat: add ability to customize Error Groups

### PR DESCRIPTION
**Pull request by jmartin4563**
## Proposed Release Notes
Added new API function called `setErrorGroupCallback`, which provides a way for you to customize the `error.group.name` attribute of errors that are captured by the agent. This attribute controls how the Errors Inbox functionality groups similar errors together.

To learn more about this function, please refer to our [example app](https://github.com/newrelic/newrelic-node-examples).

## Links
Closes NEWRELIC-7541
Example app PR: https://github.com/newrelic/newrelic-node-examples/pull/59

### CODE REVIEW

**Comment by bizob2828** [comment]
I didn't look at this PR but I just want to comment that this proposed release notes it's quite a mouthful.  can we be more succinct and link to an example?

**Comment by mrickard** [comment]
I'd say that 
> provides a way for you to customize the error.group.name attribute of errors that are captured by the agent.
would be better as 
> provides the ability to set a callback function the agent will use to assign your custom `error.group.name` to an error. In setting the callback, you can match an error's message, stack trace, or other attributes to your own Error Groups.

**Comment by jmartin4563** [comment]
> I didn't look at this PR but I just want to comment that this proposed release notes it's quite a mouthful. can we be more succinct and link to an example?

Done, PR for example app has also been submitted

**Review by mrickard — DISMISSED** [review]
Thank you!

File: api.js [hunk 1]
```diff
@@ -1739,4 +1739,49 @@ function _filterAttributes(attributes, name) {
...
+ * }
+ * newrelic.setErrorGroupCallback(myCallback)
+ */
+API.prototype.setErrorGroupCallback = function setErrorGroupCallback(callback) {
+  const metric = this.agent.metrics.getOrCreateMetric(
+    NAMES.SUPPORTABILITY.API + '/setErrorGroupCallback'
+  )
+  metric.incrementCallCount()
+
+  if (!this.shim.isFunction(callback)) {
```
Comment (jmartin4563) [inline api.js:1779]: Per standup convo, this should only accept synchronous functions for now, and we can enhance later to support async if there's a demand for it.

**Review by mrickard — APPROVED** [review]

**Pull request merged**